ACPs/113-provable-randomness/README.md

```diff
@@ -0,0 +1,141 @@
+
+To address this, a mechanism is proposed to generate verifiable, non-cryptographic random number seeds on the Avalanche platform. This method ensures uniformity while allowing developers to build more versatile applications.
+
+
+## Motivation
+
+Reliable randomness is essential for building exciting applications on Avalanche. Games, lotteries, and decentralized services all rely on unpredictable outcomes to function fairly. Randomness also fuels functionalities like unique identifiers and simulations. Without a secure way to generate random numbers within smart contracts, Avalanche applications become limited.
+
+Avalanche's traditional reliance on external oracles for randomness creates complexity and bottlenecks. These oracles inflate costs, hinder transaction speed, and are cumbersome to integrate. As Avalanche scales to more Subnets, this dependence on external systems becomes increasingly unsustainable.
+
+A solution for verifiable random number generation within Avalanche solves these problems. It provides fair randomness functionality across the chains, at no additional cost. This paves the way for a more efficient Avalanche ecosystem.
+
+## Specification
+
+### Changes Summary
+
+Existing avalanche protocol breaks the block building into two parts : external and internal. The external block is the SnowMan++ block, whereas the internal block is the actual virtual machine block.
+
+To support randomness, a BLS based VRF implementation is used, that would be recursively signing its own signatures as its message. Since the BLS signatures are constant, they provide a great way to construct a reliable VRF.
+
+For proposers that do not have a BLS key associated with their node, the hash of the signature from the previous round.
+
+In order to bootstrap the signatures chain, a missing signature would be replaced with a byte slice that is the hash product of a verifiable and trustable seed.
+
+The changes proposed here would affect the way a block is being validated. Therefore, when this change gets implemented, it needs to be deployed as a hard fork.
+
+### Changes Details
+
+#### Step 1. Adding BLS signature to proposed blocks
+
+```golang
+type statelessUnsignedBlock struct {
+	…
+	signedParentBlockSig    []byte `serialize:”true”`
+}
+```
+
+#### Step 2. Populate signature
+
+When a block proposer attempts to build a new block, it would need to use the parent block as a reference.
+
+Populating the `signedParentBlockSig` would following this logic:
+
+1. The current proposer has a BLS key
+   
+	a. If the parent block has a valid `signedParentBlockSig` signature, that signature would be signed using the proposer’s BLS key.
+
+	b. If the parent block does not have a valid `signedParentBlockSig` signature, the proposer would sign the bootStrappingBlockSignature with its BLS key ( see below )
```

Comment:
We should clarify whether the hash of:

	```
	+-------------------------+----------+------------+
	|  prefix :               | [8]byte  | "rng-derv" |
	+-------------------------+----------+------------+
	|  signedParentBlockSig : | [48]byte |  48 bytes  |
	+-------------------------+----------+------------+
	```

constitutes a "valid `signedParentBlockSig` signature" since it isn't a signature, it's a hash.

Reply:
I replaced the wording of "valid" with empty/non-empty. I think it would better describe the usage.